test/jest_integration/basic_auth.test.ts

```diff
@@ -136,4 +171,46 @@ describe('start kibana server', () => {
     const cookie = extractAuthCookie(response);
     expect(cookie.split('=')[1]).toBeFalsy();
   });
+
+  it('anonymous auth', async () => {
+    const response = await kbnTestServer.request
+      .get(root, '/auth/anonymous')
+      .unset(AUTHORIZATION_HEADER_NAME);
+    expect(response.status).toEqual(200);
+    expect(response.body.username).toEqual('opendistro_security_anonymous');
+  });
+
+  it('anonymous disabled', async () => {
+    const anonymousDisabledRoot = kbnTestServer.createRootWithSettings(
+      {
+        plugins: {
+          scanDirs: [resolve(__dirname, '../..')],
+        },
+        elasticsearch: {
+          hosts: ['https://localhost:9200'],
+          ignoreVersionMismatch: true,
+          ssl: { verificationMode: 'none' },
+          username: KIBANA_SERVER_USER,
+          password: KIBANA_SERVER_PASSWORD,
+        },
+        opendistro_security: {
+          auth: {
+            anonymous_auth_enabled: false,
+          },
+        },
+      },
+      {
+        // to make ignoreVersionMismatch setting work
+        // can be removed when we have corresponding ES version
+        dev: true,
+      }
+    );
+    await anonymousDisabledRoot.setup();
+    await anonymousDisabledRoot.start();
+    const response = await kbnTestServer.request
+      .get(anonymousDisabledRoot, '/auth/anonymous')
+      .unset(AUTHORIZATION_HEADER_NAME);
+    expect(response.status).toEqual(400);
```

Comment:
So if anonymous disable, it will return 400 bad request rather than 401 unauthorized?

Reply:
yes, this API is dedicate to anonymous auth, so if client hit this API while anonymous auth is disabled, it is a bad request.